base/abstractarray.jl

```diff
@@ -101,12 +101,20 @@ end
 
 Return `true` if the indices of `A` start with something other than 1 along any axis.
 If multiple arguments are passed, equivalent to `has_offset_axes(A) | has_offset_axes(B) | ...`.
+See also [`require_one_based_indexing`](@ref)
 """
 has_offset_axes(A) = _tuple_any(x->Int(first(x))::Int != 1, axes(A))
 has_offset_axes(A::AbstractVector) = Int(firstindex(A))::Int != 1 # improve performance of a common case (ranges)
 has_offset_axes(A...) = _tuple_any(has_offset_axes, A)
 has_offset_axes(::Colon) = false
 
+"""
+    require_one_based_indexing(A::AbstractArray)
+    require_one_based_indexing(A,B...)
+
+Throw an `ArgumentError` if the indices of any argument start with something other than one along any axis.
```

Comment:
I believe this requires Julia at least 1.1.0.

Reply:
Oh, thanks for catching that. I will check and add an appropriate compat entry.

Reply:
```suggestion
Throw an `ArgumentError` if the indices of any argument start with something other than `1` along any axis.
```